repair/repair.cc

```diff
@@ -1409,6 +1439,12 @@ static future<> do_repair_ranges(lw_shared_ptr<repair_info> ri) {
                         return make_ready_future<stop_iteration>(stop_iteration::yes);
                     }
                 });
+            }).then([ri] {
+                if (ri->reason == streaming::stream_reason::bootstrap) {
+                    _node_ops_metrics.bootstrap_nr_ranges_finished++;
+                } else if (ri->reason == streaming::stream_reason::rebuild) {
+                    _node_ops_metrics.rebuild_nr_ranges_finished++;
+                }
```

Comment:
The code indicated that maybe bootstrap and rebuild should be a label.

Does it make sense to sum them both together?

Reply:
We only allow one operation at a time. It does not make sense to sum different operations.

Reply:
We have limited number of operations, so we do not have tons of "labels". It is much easier not to use label.

Reply:
I will not insist on labels, but it does sound like you are counting the same things (i.e. ranges).
In what perspective using labels make it harder?
There is a single place in the code that you defined them so it's an additional one of code to define them.
And when using them it's pretty similar if you are looking for a metric or a metric with a label.

Reply:
I did not find a document / code comments how to use it. The example I found in cql3/query_processor.cc looks very complicated.

Reply:
you can look at `./alternator/stats.cc` though their macro use is a bit confusing.

you define a label somewhere in source file like:
```
seastar::metrics::label op("op");
```

and then `op("rebuild")` will generate a new label that  you can add to a metrics either as a parameter:
```
sm::make_gauge("bootstrap_nr_ranges_total", 
                            sm::description("Number of total ranges for bootstrap operation on this shard."),
{op("rebuild")},
bootstrap_nr_ranges_total)
```

Or using the parentheses operator:
```
sm::make_gauge("bootstrap_nr_ranges_total", 
                            sm::description("Number of total ranges for bootstrap operation on this shard."),
bootstrap_nr_ranges_total)(op("rebuild"))
```

Reply:
OK. Thanks for the tip.